Client/Frontend/AuthenticationManager/AuthenticationSettingsViewController.swift

```diff
@@ -115,28 +115,32 @@ extension RequirePasscodeSetting: PasscodeEntryDelegate {
     }
 }
 
-class TouchIDSetting: Setting {
-    private let authInfo: AuthenticationKeychainInfo?
+class AuthenticationSetting: Setting {
+    private var authInfo: AuthenticationKeychainInfo?
 
     private weak var navigationController: UINavigationController?
     private weak var switchControl: UISwitch?
 
-    private var touchIDSuccess: (() -> Void)? = nil
-    private var touchIDFallback: (() -> Void)? = nil
+    private var touchIDSuccess: (AuthenticationSetting -> Void)? = nil
+    private var touchIDFallback: (AuthenticationSetting -> Void)? = nil
+    
+    var authenticationDelegate: AuthenticationSettingDelegate!
```

Comment:
In most cases, you'll want to keep a weak ref to a delegate instead of a strong one. I think that's what you want here since you don't want the setting to have a strong ref to the delegate.